include/rsl/monad.hpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+
+#include <tl_expected/expected.hpp>
+
+#include <optional>
+
+namespace rsl {
+
+/**
```

Comment:
This all looks good. My only question is should we standardize on one doxygen style and is this it?

Reply:
Since this project currently does not generate docs I'm inclined to remove the Doxygen comments. If we keep them, I think I'd prefer an asterisk-lite solution purely for aesthetic reasons.